specs/_features/eip6110/validator.md

```diff
@@ -0,0 +1,42 @@
+# EIP-6110 -- Honest Validator
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Prerequisites](#prerequisites)
+- [Block proposal](#block-proposal)
+  - [Deposits](#deposits)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This document represents the changes to be made in the code of an "honest validator" to implement EIP-6110.
+
+## Prerequisites
+
+This document is an extension of the [Capella -- Honest Validator](../../capella/validator.md) guide.
+All behaviors and definitions defined in this document, and documents it extends, carry over unless explicitly noted or overridden.
+
+All terminology, constants, functions, and protocol mechanics defined in the updated Beacon Chain doc of [EIP-6110](./beacon-chain.md) are requisite for this document and used throughout.
+Please see related Beacon Chain doc before continuing and use them as a reference throughout.
+
+## Block proposal
+
+### Deposits
+
+The expected number of deposits MUST be changed from `min(MAX_DEPOSITS, eth1_data.deposit_count - state.eth1_deposit_index)` to the result of the following function:
```

Comment:
note to myself: I just figured out that this PR kind of deprecates `MAX_DEPOSITS` preset. `MAX_DEPOSITS` doesn't represent the upper bound of deposits anymore.